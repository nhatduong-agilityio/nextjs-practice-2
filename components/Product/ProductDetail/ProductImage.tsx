import { Box, Flex, Stack } from '@chakra-ui/react'
import React, { memo } from 'react'
import { v4 as uuidv4 } from 'uuid'
import Image from '@components/Common/Image'

interface IProps {
  image: string
  altImage: string
  listImage?: string[]
}

const ProductImage = memo(({ image, altImage, listImage }: IProps) => (
  <Flex w={['full', '50%', '50%']} flexDir={['column', 'row-reverse', 'row-reverse']}>
    <Stack w='full' style={{ margin: 0 }}>
      <Box w='full' borderRadius='5px' overflow={['unset', 'unset', 'hidden']}>
        <Image alt={altImage} src={image} height={['250', '503', '503']} width='full' />
      </Box>
    </Stack>
    <Stack
      style={{ margin: 0 }}
      flexDir={['row', 'column', 'column']}
      p={['20px', 0, 0]}
      pr={[0, '25px', '25px']}
      gap={['10px', '25px', '25px']}
    >
      {listImage?.map((item) => (
        <Box key={`${uuidv4()}`} w={['60px', '107px', '107px']} style={{ margin: 0 }}>
          <Image
            alt={altImage}
            src={item}
            height={['60px', '107px', '107px']}
            variant='primary'
            width='full'
          />
        </Box>
      ))}
    </Stack>
  </Flex>
))

ProductImage.displayName = 'ProductProductImage'

export default ProductImage
