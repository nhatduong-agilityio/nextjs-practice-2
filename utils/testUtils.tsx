import { ChakraProvider } from '@chakra-ui/react'
import { AuthProvider } from '@store/providers/AuthProvider'
import { LoadingProvider } from '@store/providers/LoadingProvider'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import { customTheme } from '@themes/index'
import type { ReactElement } from 'react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={customTheme}>
      <LoadingProvider>
        <AuthProvider>{children}</AuthProvider>
      </LoadingProvider>
    </ChakraProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
