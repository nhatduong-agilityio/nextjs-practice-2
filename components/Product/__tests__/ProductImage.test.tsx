// Utils
import { render } from '@utils/testUtils'

// Components
import { ProductsMock } from '@mock/dataMock'
import type { TProduct } from '@matched-types/product'
import ProductImage from '../ProductDetail/ProductImage'

describe('ProductImage render', () => {
  const productMock = ProductsMock[0] ? ProductsMock[0] : ({} as TProduct)

  it('Should ProductImage match Snapshot', () => {
    const { container } = render(
      <ProductImage
        image={productMock.images.image}
        altImage={productMock.imageAlt}
        listImage={productMock.images.listImage}
      />,
    )

    expect(container).toMatchSnapshot()
  })
})
