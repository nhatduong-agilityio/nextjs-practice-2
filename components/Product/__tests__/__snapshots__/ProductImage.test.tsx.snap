// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage render Should ProductImage match Snapshot 1`] = `
<div>
  <div
    class="css-139259g"
  >
    <div
      class="chakra-stack css-ppmla1"
      style="margin: 0px;"
    >
      <div
        class="css-m1crd5"
      >
        <div
          class="css-6q4dsv"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="Picture of Full set Grey Sofa With Pillows"
              data-nimg="fill"
              decoding="async"
              sizes="100vw"
              src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
            />
          </span>
        </div>
      </div>
    </div>
    <div
      class="chakra-stack css-8avhs7"
      style="margin: 0px;"
    >
      <div
        class="css-lwsiv3"
        style="margin: 0px;"
      >
        <div
          class="css-1tvx95n"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="Picture of Full set Grey Sofa With Pillows"
              data-nimg="fill"
              decoding="async"
              sizes="100vw"
              src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
            />
          </span>
        </div>
      </div>
      <div
        class="css-lwsiv3"
        style="margin: 0px;"
      >
        <div
          class="css-1tvx95n"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="Picture of Full set Grey Sofa With Pillows"
              data-nimg="fill"
              decoding="async"
              sizes="100vw"
              src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
            />
          </span>
        </div>
      </div>
      <div
        class="css-lwsiv3"
        style="margin: 0px;"
      >
        <div
          class="css-1tvx95n"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="Picture of Full set Grey Sofa With Pillows"
              data-nimg="fill"
              decoding="async"
              sizes="100vw"
              src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
            />
          </span>
        </div>
      </div>
      <div
        class="css-lwsiv3"
        style="margin: 0px;"
      >
        <div
          class="css-1tvx95n"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="Picture of Full set Grey Sofa With Pillows"
              data-nimg="fill"
              decoding="async"
              sizes="100vw"
              src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
